analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -0,0 +1,108 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer
+
+import ch.frankel.slf4k.*
+
+import com.here.ort.analyzer.managers.Unmanaged
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.AnalyzerConfiguration
+import com.here.ort.model.AnalyzerResultBuilder
+import com.here.ort.model.AnalyzerRun
+import com.here.ort.model.Environment
+import com.here.ort.model.OrtResult
+import com.here.ort.model.ProjectAnalyzerResult
+import com.here.ort.model.Repository
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.utils.log
+
+import java.io.File
+
+class Analyzer {
+    fun analyze(config: AnalyzerConfiguration, absoluteProjectPath: File,
+                packageManagers: List<PackageManagerFactory<PackageManager>> = PackageManager.ALL,
+                packageCurationsFile: File? = null
+    ): OrtResult {
+        // Map of files managed by the respective package manager.
+        val managedDefinitionFiles = if (packageManagers.size == 1 && absoluteProjectPath.isFile) {
+            // If only one package manager is activated, treat the given path as definition file for that package
+            // manager despite its name.
+            mutableMapOf(packageManagers.first() to listOf(absoluteProjectPath))
+        } else {
+            PackageManager.findManagedFiles(absoluteProjectPath, packageManagers).toMutableMap()
+        }
+
+        val hasDefinitionFileInRootDirectory = managedDefinitionFiles.values.flatten().any {
+            it.parentFile.absoluteFile == absoluteProjectPath
+        }
+
+        if (managedDefinitionFiles.isEmpty() || !hasDefinitionFileInRootDirectory) {
+            managedDefinitionFiles[Unmanaged] = listOf(absoluteProjectPath)
+        }
+
+        if (log.isInfoEnabled) {
+            // Log the summary of projects found per package manager.
+            managedDefinitionFiles.forEach { manager, files ->
+                // No need to use curly-braces-syntax for logging here as the log level check is already done above.
+                log.info("$manager projects found in:")
+                log.info(files.joinToString("\n") {
+                    "\t${it.toRelativeString(absoluteProjectPath).let { if (it.isEmpty()) "." else it }}"
+                })
+            }
+        }
+
+        val vcs = VersionControlSystem.getCloneInfo(absoluteProjectPath)
+        val analyzerResultBuilder = AnalyzerResultBuilder()
+
+        // Resolve dependencies per package manager.
+        managedDefinitionFiles.forEach { manager, files ->
+            val results = manager.create(config).resolveDependencies(absoluteProjectPath, files)
+
+            val curatedResults = packageCurationsFile?.let {
+                val provider = YamlFilePackageCurationProvider(it)
```

Comment:
This re-parses the yaml file on every iteration of the `forEach`, right? Does it make sense to move `provider` out of the loop? It could be a nullable value and line 78 be changed to `val curatedResults = provider?.let { ... }`.

Reply:
This code was just moved to another file, I would address this in a separate PR.